Source/processcontainers/implementations/DobbyImplementation/DobbyImplementation.cpp

```diff
@@ -0,0 +1,304 @@
+
+        if (!runningContainers.empty())
+        {
+            for (const std::pair<int32_t, std::string>& c : runningContainers)
+            {
+                if(c.second == name){
+                    // found the container, now try stopping it...
+                    TRACE_L1("destroying container: %s ", name.c_str());
+                    bool stoppedSuccessfully = mDobbyProxy->stopContainer(c.first, true);
+                    if (!stoppedSuccessfully) {
+                        TRACE_L1("Failed to destroy container, internal Dobby error. id: %s descriptor: %d", name.c_str(), c.first);
+                    }
+                    
+                    break;
+                }
+            }
+        }
+    }
+
+    bool DobbyContainerAdministrator::ContainerNameTaken(const string& name)
+    {
+        bool result = false;
+        
+        // Get the running containers from Dobby
+        std::list<std::pair<int32_t, std::string>> runningContainers;
+        runningContainers = mDobbyProxy->listContainers();
+
+        // Build the response if containers were found
+        if (!runningContainers.empty())
+        {
+            for (const std::pair<int32_t, std::string>& c : runningContainers)
+            {
+                if(c.second == name){
+                    TRACE_L1("container %s already running...", name.c_str());
+                    result = true;
+                    break;
+                }
+            }
+        }
+
+        return result;
+    }
+
+    // Container
+    // ------------------------------------
+    DobbyContainer::DobbyContainer(const string& name, const string& path, const string& logPath)
+        : DobbyContainerMixins(name)
+        , _refCount(1)
+        , _name(name)
+        , _path(path)
+        , _logPath(logPath)
+        , _pid()
+    {
+    }
+
+    DobbyContainer::~DobbyContainer()
+    {
+        auto& admin = static_cast<DobbyContainerAdministrator&>(DobbyContainerAdministrator::Instance());
+
+        if (admin.ContainerNameTaken(_name) == true) {
+            Stop(Core::infinite);
+        }
+
+        admin.RemoveContainer(this);
+    }
+
+    const string& DobbyContainer::Id() const
+    {
+        return _name;
+    }
+
+    uint32_t DobbyContainer::Pid() const
+    {
+        uint32_t returnedPid = 0;
+        auto& admin = static_cast<DobbyContainerAdministrator&>(DobbyContainerAdministrator::Instance());
+        
+        if (_pid.IsSet() == false) {
+            
+            
+            std::string containerInfoString = admin.mDobbyProxy->getContainerInfo(_descriptor);
+
+            if (containerInfoString.empty())
+            {
+                TRACE_L1("Failed to get info for container %s", _name.c_str());
+            }else{
+                // Dobby returns the container info as JSON, so parse it
+                JsonObject containerInfoJson;
+                WPEFramework::Core::OptionalType<WPEFramework::Core::JSON::Error> error;
```

Comment:
uhhh now you lost me :-) here you are using the Thunder JSON parsing stuff... So why include this jsoncpp ?